tests/crew_test.py

```diff
@@ -722,6 +745,38 @@ def test_agent_usage_metrics_are_captured_for_hierarchical_process():
     }
 
 
+@pytest.mark.vcr(filter_headers=["authorization"])
+def test_hierarchical_crew_creation_tasks_with_agents():
+    """
+    Agents are not required for tasks in a hierarchical process but sometimes they are still added
+    This test makes sure that the manager still delegates the task to the agent even if the agent is passed in the task
+    """
+    from langchain_openai import ChatOpenAI
+
+    task = Task(
+        description="Write one amazing paragraph about AI.",
+        expected_output="A single paragraph with 4 sentences.",
+        agent=writer,
+    )
+
+    crew = Crew(
+        tasks=[task],
+        agents=[writer],
+        process=Process.hierarchical,
+        manager_llm=ChatOpenAI(model="gpt-4o"),
+    )
+
+    assert crew.process == Process.hierarchical
+    assert crew.manager_llm is not None
+    assert crew.tasks[0].agent == writer
+    result = crew.kickoff()
```

Comment:
Right now, this isn't testing to make sure that during the `_run_hierarchical_process` that the manager agent's tools are only set to the writer when executing the task.

We would have to do some mocking to actually make this test valid.

Reply:
i think i can do another assert to look into crew.task[0].tools == delegate(Agent=[writer]) or something like that